frontend/src/components/file-explorer/FileExplorer.test.tsx

```diff
@@ -30,18 +30,24 @@ describe("FileExplorer", () => {
   });
 
   it("should get the workspace directory", async () => {
-    const { getByText } = renderWithProviders(<FileExplorer />);
+    const { getByText } = renderWithProviders(<FileExplorer />, {
+      preloadedState: {
+        agent: {
+          curAgentState: AgentState.RUNNING,
+        },
+      },
+    });
 
     await waitFor(() => {
       expect(getByText("folder1")).toBeInTheDocument();
-      expect(getByText("file2.ts")).toBeInTheDocument();
+      expect(getByText("file1.ts")).toBeInTheDocument();
     });
-    expect(listFiles).toHaveBeenCalledTimes(2); // once for root, once for folder1
+    expect(listFiles).toHaveBeenCalledTimes(1);
   });
 
   it.todo("should render an empty workspace");
 
-  it.only("should refetch the workspace when clicking the refresh button", async () => {
```

Comment:
Wow! How long has this been there!? 😱

Reply:
😭 it's been for a while since it failed in the GitHub Action. 

BTW, the unit tests made me feel it was generated by AI 😅

Reply:
No I've written them 😅 (well with GitHub Copilot). They were written before the component, and both the tests and implementation were meant to be as simple as possible. They have changed somewhat since I've written them so some differences since first written.

Did you find something missing/wrong/strange with them?